File: Introductory Problems/Creating Strings/creating_strings.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long int

vector<string> ans;

void helper(string s, int idx, int n){
	if(idx == n){
		ans.push_back(s);
		return;
	}
	unordered_set<char> st;
	for(int i=idx;i<n;i++){
		if(st.find(s[i]) != st.end()) continue;
		st.insert(s[i]);
		swap(s[idx], s[i]);
		helper(s, idx+1, n);
		swap(s[idx], s[i]);
	}
	return;
}

void solve(){
	string s;
	cin >> s;
	int n = s.size();

	helper(s, 0, n);
	sort(ans.begin(), ans.end());

	cout<<ans.size()<<endl;
	for(int i=0;i<ans.size();i++){
		cout<<ans[i]<<endl;
	}
	return;
	
}

int main(){
	long long testCases;
	testCases = 1;
	// cin>>testCases;
	while(testCases--){
		solve();
	}
	return 0;
}

File: Introductory Problems/Creating Strings/creatingStrings.java
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Scanner;
import java.util.Set;

public class creatingStrings {
    static List<String> ans = new ArrayList<>();

    static void helper(StringBuilder s, int idx, int n) {
        if (idx == n) {
            ans.add(s.toString());
            return;
        }
        Set<Character> st = new HashSet<>();
        for (int i = idx; i < n; i++) {
            if (st.contains(s.charAt(i))) continue;
            st.add(s.charAt(i));
            swap(s, idx, i);
            helper(s, idx + 1, n);
            swap(s, idx, i);
        }
    }

    static void swap(StringBuilder s, int i, int j) {
        char temp = s.charAt(i);
        s.setCharAt(i, s.charAt(j));
        s.setCharAt(j, temp);
    }

    static void solve() {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.next();
        int n = s.length();

        helper(new StringBuilder(s), 0, n);
        ans.sort(null);

        System.out.println(ans.size());
        for (String permutation : ans) {
            System.out.println(permutation);
        }
        scanner.close();
    }

    public static void main(String[] args) {
        long testCases = 1;
        while (testCases-- > 0) {
            solve();
        }
    }
}
